### PR TITLE
Feat/Read Only View for trips in Discover.

### LINKED ISSUE #277: Add Read only view of trip in Discover Page
Right now we cannot see trip details in the discover page, we should allow users to see the given activities of that trip.

### PR DESCRIPTION
**Pull request by 1andreea26**
## Summary

This PR introduces the possibility to view a trip from the Discover feed in Read Only View Mode. 
When the user presses on `View Details` button on a trip displayed in the Discover feed, they are redirected to the ByDay screen, without the possibility to add a new activity/ask the AI assistant. Moreover, the TopTabs bar now has only 2 options: `Schedule` (ByDay screen and Weekly screen) or `Activities` ( `Photos `and `Settings` screens have been disabled).
 - For the `Schedule` screens,  the possibility to add a new activity or ask the AI assistant has been disabled, and if the user presses on an Activity -> `ActivitiesForOneDay` screen displays the activity, with the possibility to add an activity or edit/delete disabled.
 - For the `Activities` screen, the possibility to add a new activity or edit/delete an existent one has been disabled.
## Changes

- **Screens/UI:**  Schedule Screens (ByDay Screen and WeeklyView Screen) and the Activities Screen now have a `isReadOnly` parameter (set to false by default) which disables the add an activity button, respectively the edit/delete buttons for existent activities.
- **Logic/Controllers:** The Navigation State in Navigation Actions now has a variable isReadOnlyView, used to track 
which screens are accessible and show the corresponding tabs in the TopTab bar (disabling 2 of the possible screens accessible by the TopTab bar).
- **Bug Fixes/Improvements:** Addresses #277. 

## Checklist

- [X] This PR resolves #277.
- [X] Tests have been added for new functionality.
- [X] Screenshots or UI changes have been attached (if applicable).
- [X] Documentation has been updated (if applicable).

##  Testing

- **Manual Testing:**
    - [X] Tested on emulator / physical device.
- **Unit Tests**
    - [X] Added units tests for this

##  Related Issues/Tickets

Closes #277 

## Screenshots (if applicable)

Screenshot 2024-12-10 at 15 48 55
Screenshot 2024-12-10 at 15 49 14
Screenshot 2024-12-10 at 15 49 37
Screenshot 2024-12-10 at 15 49 55

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt
Comment (ioanajianu) [inline app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt:59]: nit: for consistency, call TripCard with `tripsViewModel = tripsViewModel, navigationActions = navigationActions` (the same way we pass `trip = trips[page]`

File: app/src/androidTest/java/com/android/voyageur/ui/components/PlaceSearchWidgetTest.kt
Comment (ioanajianu) [inline app/src/androidTest/java/com/android/voyageur/ui/components/PlaceSearchWidgetTest.kt:94]: nit: mention somewhere in the PR description that you removed this test because it was flaky

File: app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt
Comment (ioanajianu) [inline app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt:94]: let's avoid duplicated code by doing something like 
`val tabsToDisplay = if (...) readOnlyTabs else tabs`
`tabsToDisplay.forEachIndexed...`

File: app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt
Comment (ioanajianu) [inline app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt:103]: It's not really clear to me why we pass isReadOnlyView as a parameter when we already pass navigationActions which has the readOnlyView property?

File: app/src/androidTest/java/com/android/voyageur/ui/trip/activities/ActivitiesForOneDayScreenTest.kt
Comment (ioanajianu) [inline app/src/androidTest/java/com/android/voyageur/ui/trip/activities/ActivitiesForOneDayScreenTest.kt:1]: Can you also add a test that checks that the AddActivity button is not displayed when read mode is on?

File: app/src/androidTest/java/com/android/voyageur/ui/search/SearchScreenTest.kt
Comment (ioanajianu) [inline app/src/androidTest/java/com/android/voyageur/ui/search/SearchScreenTest.kt:159]: Can you also check that navigationActions navigate to TOP_TABS

**Review by ioanajianu — CHANGES_REQUESTED** [review]
Thanks Andreea for implementing this! Looks good, I just left a few comments. Let me know what you think!

File: app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt
Comment (1andreea26) [reply app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt:59]: I changed this, thanks for the suggestion!

File: app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt
Comment (1andreea26) [reply app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt:94]: You're right, it's better this way. I changed it.

File: app/src/androidTest/java/com/android/voyageur/ui/components/PlaceSearchWidgetTest.kt
Comment (1andreea26) [reply app/src/androidTest/java/com/android/voyageur/ui/components/PlaceSearchWidgetTest.kt:94]: After merging the changes from main, I see that someone fixed the test, so I re-added it fixed.

**Comment by 1andreea26** [comment]
> Thanks Andreea for implementing this! Looks good, I just left a few comments. Let me know what you think!

Thanks for the review Ioana! I just implemented your suggestions and answered the comments.

File: app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt
Comment (BogdanCDev) [inline app/src/main/java/com/android/voyageur/ui/search/DiscoverContent.kt:73]: You should update the documentation for this Composable.

**Comment by BogdanCDev** [comment]
Great job with this implementation! The documentation looks great, aside from a composable that you probably just missed to update, and also the line coverage is above the target. Also, the overall implementation looks really solid! Great work!

File: app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt
Comment (ioanajianu) [reply app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt:97]: I see what you mean. My thought was that by not passing it as a parameter we can avoid situations where `navigationActions.getNavigationState().isReadOnlyView` is true, but the parameter is false (or the other way around).
However, I understand how this would make testing more difficult. 
I'll approve this PR for now as this choice doesn't impact the app, and I'll leave it up to you. If you'd like, we can look more into this next sprint, I think I worked on some similar tests for the current tab field of the `navigtionActions` :)

**Review by ioanajianu — APPROVED** [review]
Thanks a lot Andreea for implementing this change and taking into account my suggestions! I tested this feature and it works well, great work! Also, the test coverage is higher than 80%! LGTM!

**Comment by 1andreea26** [comment]
> Thanks a lot Andreea for implementing this change and taking into account my suggestions! I tested this feature and it works well, great work! Also, the test coverage is higher than 80%! LGTM!

Thanks for the review Ioana!

**Pull request merged**
